### ISSUE TITLE
MA0179's "compliant - accessing attribute properties" setup could be refined.

### ISSUE THREAD

**Issue by mjr4077au** [meziantou/Meziantou.Analyzer#1219]
### Version of the Meziantou.Analyzer NuGet package

Latest

### Rule Identifier

MA0179

### Target Framework

net472

### C# Language version

C# 14

### Description

The provided code in "Reproduction Steps" triggers MA0179 whereas the [docs](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0179.md) indicate this shouldn't occur: https://github.com/meziantou/Meziantou.Analyzer/blob/c99bc8e626b638c2d89272ff08719fe66997511b/docs/Rules/MA0179.md?plain=1#L36-L41

If I'm storing off the result in a variable, I'd expect that the analyser understands I'm intending to use it if it's not null. It does work if I reverse the if statement logic, however it's a normal pattern that guard clauses throw rather than early return.

### Reproduction Steps

Minimal code:

```c#
if (type.GetCustomAttribute<CmdletAttribute>() is not CmdletAttribute cmdletAttribute)
{
    throw new ArgumentException($"Type '{type.FullName}' does not have a CmdletAttribute.", nameof(type));
}
return $"{cmdletAttribute.VerbName}-{cmdletAttribute.NounName}";
```

**Comment by meziantou** [comment]
Thanks for reporting it! The check for patterns was not strict enough. The next version should fix it.

**Comment by mjr4077au** [comment]
You're truly a legend! Thanks so much 🤘

<img width="500" height="275" alt="Image" src="https://github.com/user-attachments/assets/b22ea3c5-9b22-4841-b8ae-2c5a0b849d16" />
